backup-stores/azure/src/test/java/io/camunda/zeebe/backup/azure/AzureBackupStoreIT.java

```diff
@@ -78,7 +92,41 @@ void backupShouldExistAfterStoreIsClosed(final Backup backup) {
   }
 
   @Test
-  void cannotDeleteUploadingBlock() {
-    // TODO: when delete feature is done
+  public void cannotDeleteUploadingBlock() throws IOException, InterruptedException {
+    // given
+    final var tempDir = Files.createTempDirectory("backup");
+    Files.createDirectory(tempDir.resolve("segments/"));
+    final var seg1 = Files.createFile(tempDir.resolve("segments/segment-file-1"));
+
+    Files.write(seg1, RandomUtils.nextBytes(50 * 1024 * 1024));
+
+    final Backup largeBackup =
+        new BackupImpl(
+            new BackupIdentifierImpl(1, 2, 3),
+            new BackupDescriptorImpl(Optional.empty(), 4, 5, "test"),
+            new NamedFileSetImpl(Map.of()),
+            new NamedFileSetImpl(Map.of("segment-file-1", seg1)));
+
+    // when
+    // Takes long to save 50MB
```

Comment:
:+1: Good idea to add a test for this

:x: I suspect that this will be flaky, we shouldn't rely on network throughput and timing for this.
Since this test is specifically for Azure, maybe we can use internal knowledge and just manipulate the manifest such that it is in-progress?